### PR TITLE
fix list with default

### PR DESCRIPTION
**Pull request by eellison**
Fixing list with default

### CODE REVIEW

**Review by driazati — APPROVED** [review]

File: torch/csrc/jit/register_special_ops.cpp [hunk 1]
```diff
@@ -35,20 +35,35 @@ RegisterOperators reg({
...
+        IValue single_val;
+        IValue defaults;
+        pop(stack, single_val, defaults);
+        push(stack, single_val);
+        return 0;
+      }),
+    Operator(
+        "aten::list_with_default(int?[] list, int[] defaults) -> int[]",
         [](Stack& stack) {
           autograd::profiler::RecordFunction record("sizes");
```
Comment (driazati) [inline torch/csrc/jit/register_special_ops.cpp]: This can be removed

**Review by driazati — APPROVED** [review]

File: test/test_jit.py [hunk 2]
```diff
@@ -7680,6 +7680,37 @@ def test_no_type():
                 # type: () -> int
                 return torch.jit._unwrap_optional(None)
 
+    def test_list_of_optionals(self):
+        @torch.jit.script
```
Comment (wanchaol) [inline test/test_jit.py]: this test isn't really necessary right?

**Review by wanchaol — APPROVED** [review]

File: torch/csrc/jit/register_special_ops.cpp (hunk 1 above)
Comment (eellison) [reply torch/csrc/jit/register_special_ops.cpp]: even if it's not used in current context, i think it's good to match the semantics of the original function

File: test/test_jit.py (hunk 2 above)
Comment (eellison) [reply test/test_jit.py]: Technically it's already being tested through torch/nn, but I think it's good to have modular tests

Comment (wanchaol) [reply test/test_jit.py]: Can you make these two tests to test_optional_conversion to keep a single place for testing those functionalities?

**Pull request closed without merging**